src/combinatorics/quiver.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2021 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+import tactic
+/-!
+# Quivers
+
+This module defines quivers. A quiver `G` on a type `V` of vertices assigns to every
+pair `a b : V` of vertices a type `G a b`, thought of as edges from `a` to `b`. This
+is a very permissive notion of graph.
+-/
+
+-- We use the same universe order as in category theory.
+universes v u
+
+/-- A quiver `G` on a type `V` of vertices assigns to every pair `a b : V` of vertices
+    a type `G a b`, thought of as edges from `a` to `b`. -/
+def quiver (V : Type u) := V → V → Type v
+
+/-- The default quiver on `V` is the empty quiver. -/
+instance {V} : inhabited (quiver.{v} V) :=
+⟨λ a b, pempty⟩
+
+/-- A subquiver `H` of `G` picks out subset `H a b` of the edges from `a` to `b`
+    for every pair of vertices `a b`. -/
+def subquiver {V} (G : quiver V) :=
+Π a b : V, set (G a b)
```

Comment:
Isn't a subquiver a subset of `V` along with subsets of edges between vertices in that subset?  That's important for decomposing a quiver into connected components.

Reply:
Good point. I still think we'll want this "subset of edges" as a separate notion, but we should maybe call it something other than `subquiver`.

Reply:
I decided to call this `wide_subquiver`, in analogy with "wide subcategory".